src/buffer/mod.rs

```diff
@@ -4,15 +4,65 @@ mod immutable;
 mod iterator;
 
 use crate::ffi::InternalArrowArray;
+use std::ops::Deref;
 
 pub(crate) enum BytesAllocator {
     InternalArrowArray(InternalArrowArray),
 
     #[cfg(feature = "arrow")]
     Arrow(arrow_buffer::Buffer),
 }
+pub(crate) type BytesInner<T> = foreign_vec::ForeignVec<BytesAllocator, T>;
 
-pub(crate) type Bytes<T> = foreign_vec::ForeignVec<BytesAllocator, T>;
+/// Bytes representation.
```

Comment:
Makes the struct public, so that it can be passed as argument, but keeps internals hidden.